test/units/testsuite-36.sh

```diff
@@ -26,25 +27,30 @@ testUnitFile="/run/systemd/system/$testUnit"
 testUnitNUMAConf="$testUnitFile.d/numa.conf"
 
 # Sleep constants (we should probably figure out something better but nothing comes to mind)
-journalSleep=5
 sleepAfterStart=1
 
 # Journal cursor for easier navigation
 journalCursorFile="jounalCursorFile"
 
 startStrace() {
-    coproc strace -qq -p 1 -o $straceLog -e set_mempolicy -s 1024 $1
+    coproc strace -qq -p 1 -o "$straceLog" -e set_mempolicy -s 1024 ${1:+"$1"}
     # Wait for strace to properly "initialize"
     sleep $sleepAfterStart
 }
 
 stopStrace() {
-    kill -s TERM $COPROC_PID
+    if [[ ! -v COPROC_PID ]]; then
+        return
+    fi
+
+    local PID=$COPROC_PID
+    kill -s TERM "$PID"
     # Make sure the strace process is indeed dead
-    while kill -0 $COPROC_PID 2>/dev/null; do sleep 0.1; done
+    while kill -0 "$PID" 2>/dev/null; do sleep 0.1; done
```

Comment:
Why make this local? $COPROC_PID should not change.

Reply:
`$COPROC_PID` has a really annoying behavior, where it disappears when the coproc-ed process exits. This, in combination with `-set u` leads to a sort of unexpected error:

```
$ set -u
$ coproc sleep 10
[1] 1057765
$ while kill -0 $COPROC_PID; do echo sleeping; sleep 1; done
sleeping
sleeping
sleeping
sleeping
sleeping
sleeping
sleeping
[1]+  Done                    coproc COPROC sleep 10
bash: COPROC_PID: unbound variable
```

*just bash things*

Reply:
Oh, it's "magic". Yikes.